### PR TITLE
Create webb_extended_mixer_milestone_2.md

### PR DESCRIPTION
**Pull request by drewstone**
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [ ] The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out for this milestone.
- [x] This pull request is being made by the same account as the accepted application.
- [x] In case of acceptance, the payment will be transferred to the BTC/ETH payment address provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://github.com/w3f/Grants-Program/blob/master/docs/milestone-deliverables-guidelines.md).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/422

### CODE REVIEW

File: deliveries/webb_extended_mixer_milestone_2.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Milestone Delivery :mailbox:
+
+**The [invoice form :pencil:](https://docs.google.com/forms/d/e/1FAIpQLSfmNYaoCgrxyhzgoKQ0ynQvnNRoTmgApz9NrMp-hd8mhIiO0A/viewform) has been filled out correctly for this milestone and the delivery is according to the official [milestone delivery guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/milestone-deliverables-guidelines.md).**  
+
+* **Application Document:** https://github.com/w3f/Grants-Program/pull/422
```
Comment (alxs) [inline deliveries/webb_extended_mixer_milestone_2.md:5]:
```suggestion
* **Application Document:** https://github.com/w3f/Grants-Program/blob/master/applications/MIXERv2.md
```

**Comment by alxs** [comment]
Thanks for the delivery @drewstone. Could you list the same deliverables as in the application and provide links to precisely where each of them is implemented, please?

**Comment by drewstone** [comment]
Yep @alxs just amended it.

File: deliveries/webb_extended_mixer_milestone_2.md (hunk 1 above)
Comment (alxs) [reply deliveries/webb_extended_mixer_milestone_2.md:5]:
```suggestion
* **Application Document:** https://github.com/w3f/Grants-Program/blob/master/applications/MIXERv2.md
```

File: deliveries/webb_extended_mixer_milestone_2.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
```
Comment (alxs) [inline deliveries/webb_extended_mixer_milestone_2.md:5]:
```suggestion
* **Application Document:** https://github.com/w3f/Grants-Program/blob/master/applications/MIXERv2.md
```

**Comment by alxs** [comment]
Is there any particular reason the UI requires node 14.x, or could you bump the dependency?

I ran into the following when trying to build the node with `cargo build --release -p darkwebb-standalone-node`:
```
error: package ID specification `darkwebb-standalone-node` did not match any packages
```

**Comment by alxs** [comment]
ping @drewstone

**Comment by drewstone** [comment]
Try webb-standalone-node instead of darkwebb.

**Comment by drewstone** [comment]
We’ve been using node 14. Sometimes there are issues on node 17. Likely it
can be bumped but we haven’t tried it yet.

**Comment by drewstone** [comment]
Let me know if you run into more issues @alxs

**Comment by alxs** [comment]
Hey @drewstone. I went ahead and did a preliminary evaluation of the delivery. You can find my notes [here](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/Webb-Mixer-Extended_2_alxs.md). Please get back to me when you've addressed all of the issues and just let me know if anything's unclear, or you think I missed something. Also happy to jump on a call if you have any questions.

**Comment by alxs** [comment]
@drewstone checking in here, any updates?

**Comment by drewstone** [comment]
Yes @alxs, we've been working hard to update the documentation and testing amidst the other deliverables we are working on. You can see more here:
- Docs, examples, API updates: https://github.com/webb-tools/arkworks-gadgets
- Mixer tests with API against a local node: https://github.com/webb-tools/webb.js/blob/master/tests/integration_tests/substrate/mixerTests.spec.ts
- We are updating the dApp to node 16 which has taken some time and should be done soon, docs will be added to the dApp as well.
- The relayer has undergone another large update, we are adding docs and integration tests there now too.

**Comment by alxs** [comment]
Thanks for the update @drewstone. Looks great, please let me know when I should take another look at it.

**Comment by dutterbutter** [comment]
Hey @alxs, we are ready to have you take another look at your earliest convenience.

We greatly appreciate your early feedback, and we have steadily worked on making improvements across our repos. We have increased our testing suite, readmes, documentation, and updated our dependencies since you've last reviewed. 

For an in-depth guide on how to utilize the Substrate mixer with the Webb dapp locally you may follow this [guide](https://docs.webb.tools/how-to-guides/v1/substrate-mixer/local-mixer/). 

Any and all feedback is certainly welcomed! We appreciate your patience.

**Comment by alxs** [comment]
Hey @dutterbutter, many thanks for the update and your hard work. I'll be able to take a look at this the week after next - if that's too long to wait, please just ping somebody else from the team.

**Comment by alxs** [comment]
Hey @dutterbutter, sorry for the late reply. I followed the guide you last shared and was able to run the full setup with the two nodes, relayer and dapp. The docs website looks great!

I'm still having the [same issue](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/Webb-Mixer-Extended_2_alxs.md#general-notes) I reported in my initial evaluation when trying to run the `wasm-utils` tests. I've attached the full output of `yarn test` (since it's the command provided in the instructions, I'm not sure why it would fail due to lack of priviledges) and `sudo yarn test`.

I also didn't manage to run the [mixer example](https://github.com/webb-tools/webb.js/tree/master/examples#run-the-substrate-mixer-example) in the `webb.js` repo. Also here, I've attached the output.

Furthermore, 2 integration tests in the relayer seem to be failing, as well as those in the node.

And could you please take a look at the output of `cargo clippy` on the node and address the warnings?

Lastly, please address the attribution issue - your `webb.js` repo still contains Acala code and as far as I can tell is clearly based on their [acala.js](https://github.com/AcalaNetwork/acala.js) repo, but you've currently removed their authorship everywhere and don't comply with the original [license](https://github.com/AcalaNetwork/acala.js/blob/master/LICENSE).

[webb.js mixer example.txt](https://github.com/w3f/Grant-Milestone-Delivery/files/8757159/webb.js.mixer.example.txt)
[yarn-test.txt](https://github.com/w3f/Grant-Milestone-Delivery/files/8757160/yarn-test.txt)
[sudo-yarn-test.txt](https://github.com/w3f/Grant-Milestone-Delivery/files/8757161/sudo-yarn-test.txt)
[sh run-integration.txt](https://github.com/w3f/Grant-Milestone-Delivery/files/8757162/sh.run-integration.txt)

**Comment by drewstone** [comment]
Thanks for the feedback. We will look into the licensing issue and fix what is mentioned. Some of those tests @alxs also are actually only meant to pass in CI. Specifically on `run-integration.txt`, those last tests are expected to fail locally without the right setup (which is in CI)

What node version are you using? Can you try to use Node v16?

**Comment by alxs** [comment]
Hey @drewstone, thanks for the quick reply. Regarding the integration tests, that makes sense and in that case you may want to add a disclaimer in the instructions (or remove them altogether).

I'm using Node 16 and the installation instructions didn't solve the issue, I'm still getting the output. It looks like it also needs to be installed as an npm package?

**Comment by drewstone** [comment]
Any update on the review here @alxs

**Comment by alxs** [comment]
@drewstone @dutterbutter thanks for taking care of the last few issues! As per our conversation on Element, I'm happy to accept your milestone. You can find my final notes [here](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/Webb-Mixer-Extended_2_alxs.md). Congratulations on completing the grant!

I see you haven't checked the [invoice form](https://forms.gle/LSRr7PCjBpEbKGh89) checkbox in your first comment, could you please fill it now? Let me know when you have and I'll notify the finance team to pay it.

**Comment by drewstone** [comment]
Just checked the invoice form and submitted @alxs. Let me know if there are issues.

**Comment by fededubbi** [comment]
Good afternoon,

Can you please add our complete billing address on the invoice?

Baarerstrasse 14,
6300, Zug.

Many thanks,
Regards,

Federica Dubbini

**Comment by drewstone** [comment]
Updated @fededubbi, thanks for the update.

**Comment by RouvenP** [comment]
hi @drewstone we transferred the payment today. Thanks!

**Pull request merged**